crates/iceberg/src/catalog/mod.rs

```diff
@@ -83,7 +83,7 @@ pub trait Catalog: Debug + Sync + Send {
     async fn drop_table(&self, table: &TableIdent) -> Result<()>;
 
     /// Check if a table exists in the catalog.
-    async fn stat_table(&self, table: &TableIdent) -> Result<bool>;
+    async fn table_exists(&self, table: &TableIdent) -> Result<bool>;
```

Comment:
I have no strong opinion here.

Reply:
The `<verb>_<noun>` seems to be widely used in rust standard library. But it's kind of weird to add `is_` for `exists` method, see [Path::exists](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists). So I still prefer not adding `is_` prefix.

Reply:
> But it's kind of weird to add `is_` for `exists` method, see [Path::exists](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists).

I believe this is because `exists` is an API method for `Path` itself. Our case here is more similar to [`is_file`](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file), [`is_dir`](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_dir), and [`is_absolute`](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_absolute).

Reply:
TBH I have no strong preference to either, anyway naming is the second hardest problem in software engineering. cc @Fokko What do you think?

Reply:
I'm not a native English speaker either, so bear with me here. `is_table_exists` sounds odd to me that's why I suggested `table_exists`. If it was on the table itself, I can see that you have something like: `Catalog.load('some.entity').is_view`, `Catalog.load('some.entity').is_table`. But since this method is on the Catalog, I think it makes more sense to have `table_exists`, similar to `create_table`.

Calling in @bitsondatadev

Reply:
My opinion only: Provided you can understand the basic intention, I believe making it sound like proper English isn't a concern. That's what documentation is for. To me, convention is the most important, so that if you have no documentation, ideally it's like every other function call that verifies if a table exists?

What do we call it in PyIceberg? What does Delta call it? Don't follow Hudi whatever they do, especially if it's different.

If there's no consensus `table_exists` is just fine.

Reply:
`table_exists` sounds the most natural to me. This is what PyIceberg uses as well.

https://github.com/apache/iceberg-python/blob/main/pyiceberg%2Fcatalog%2F__init__.py#L649

Reply:
Agreed, also reading back to the previous comments, `<verb>_<noun>` is always a tricky one to keep consistent when you switch from a command like `get_table` or `put_metadata`, etc.. to asking a question about the state as it changes the structure in English. So in most practices, anything that is returning a boolean gets special treatment.

More importantly, let's just do what we've done in other APIs to the best of our ability.

Reply:
Thanks everyone for this discussion, I think We have reach consensus here, let's move!